### ISSUE TITLE
Scaling an app down after a node goes down doesn't work

### ISSUE THREAD

**Issue by ruippeixotog** [mesosphere/marathon#4059]
I apologize in advance for the slightly vague title, but I couldn't think of a better one. Consider the following scenario:

- Marathon has 3 nodes;
- An app has 3 running instances, constrained to run in unique hostnames;
- One of the Marathon nodes goes down unexpectedly.

The app enters the "Waiting" state, as it doesn't have the resources to relocate the lost instance:

![image](https://cloud.githubusercontent.com/assets/613493/16527555/7c8a8470-3faf-11e6-9ac5-7f606219cde5.png)

We're developing a very basic autoscaling service that dynamically scales an app according to the number of available nodes. However, when we try to scale the app down to 2 instances in the previously mentioned state (either via the Web UI or via the API), the app remains in the "Waiting" state:

![image](https://cloud.githubusercontent.com/assets/613493/16527649/f64358be-3faf-11e6-8815-ca25b7789dce.png)

No deployments are in progress, but the `/v2/queue` endpoint returns:

```js
{
  "queue": [
    {
      "app": {
        "id": "...",
        "instances": 3,
        // (...)
      },
      "count": 1,
      "delay": {
        "timeLeftSeconds": 0,
        "overdue": true
      }
    }
  ]
}
```

I believe this is because Marathon is trying to do a rolling update. It doesn't seem to make sense however, as scaling down an app without changing anything else usually does not trigger a rolling update. If this is not a bug, can anyone explain to me why this happens and suggest an alternative approach to scale the app without affecting the currently running instances?

**Comment by meichstedt** [comment]
@ruippeixotog Thanks for reporting this, and I think the title is just fine ;)

Which Marathon version are you using? Can you provide logs for this? The deployment steps are logged so it would be interesting to see what's going on there. It should definitely just be a ScaleApplicationStep.

The LaunchQueue obviously wasn't updated and I'm wondering why there's no deployment in progress.

Seems related to #4039

**Comment by ruippeixotog** [comment]
I'm using Marathon 1.1.1 with Mesos 0.28.2.

Can you specify which logs do you need? I tried filtering useful messages from `/var/log/syslog`, but it doesn't seem to have useful information... A filtered syslog example is [here](http://pastebin.com/57UYUz9K), where I terminated a node at 16:48:00 and scaled the application (named `my-task`) down at 16:51:00. If you want I can try the case with different Marathon logging settings.

**Comment by ruippeixotog** [comment]
Hi there. Is there any update on this? I can still provide log files if you tell me where to find them. Cheers!

**Comment by unterstein** [comment]
Hi @ruippeixotog,

I think this is solved through #4039. Is it possible for you to test the 1.3.0-RC5 version of Marathon to verify if this issue is solved for you? If it is not possible to upgrade, please comment on #4232 that you need a back port. If you upgrade and this is still an issue please re-open the ticket :-)

Thanks
Johannes

**Comment by ruippeixotog** [comment]
Hi @unterstein, I finally had the chance to test this in Marathon 1.3.0. Unfortunately, I am experiencing the exact same issue... By executing the steps I described above I reach the same inconsistent state.

**Comment by unterstein** [comment]
Hi @ruippeixotog,

I added an integration test reproducing exactly your issue (currently in review: https://phabricator.mesosphere.com/D54). The good news: Your issue should be solved with 1.3.1 :)

Thanks
Johannes
